### PR TITLE
TOML files update

### PR DESCRIPTION
**Pull request by erick-xanadu**
**Context:** We would like Catalyst to optimize the first instance of `qml.StatePrep` by just setting the state vector in PennyLane lightning.

**Description of the Change:** To achieve this, we set a flag in the TOML files where it denotes whether the value for the flag `skip_initial_state_prep` used during decomposition. This flag is already in PennyLane and will skip the initial state preparation.

There are other ways to achieve this but this is the least invasive way to do so. Some other alternatives that were discussed:

1. Adding StatePrep to the list of supported operations: this is not a good idea as it would lead StatePrep to not be decomposed ever.
2. Creating a new operation and map StatePrep directly into this new operation only in the context of Catalyst. This would lead to setting this new `qml` Operation in the list of supported operations on the device, even though it would never be seen except when using with Catalyst.

The way this was achieved was not to create a new `qml` Operation, only an JAXPR/MLIR operation and bind `StatePrep` to this JAXPR operation and lower it through MLIR. So, no new `qml` operations.

**Benefits:** Optimization

**Possible Drawbacks:** More flags. No one likes flags.

**Related GitHub Issues:** Will be followed by this https://github.com/PennyLaneAI/catalyst/pull/955 pull request in Catalyst. 

[sc-69069]

### CODE REVIEW

**Comment by erick-xanadu** [comment]
@dime10 , please let me know if you want a different flag name. I think this is appropriate.

File: pennylane_lightning/lightning_kokkos/lightning_kokkos.toml [hunk 1]
```diff
@@ -112,3 +112,6 @@ dynamic_qubit_management = false
 # in a single execution
 non_commuting_observables = true
 
+# Whether the compilation pass should optimize the first
+# state preparation as a SetState operation.
+skip_initial_state_prep = false
```
Comment (maliasadi) [inline pennylane_lightning/lightning_kokkos/lightning_kokkos.toml:117]: This should be "true" for both [LK](https://github.com/PennyLaneAI/pennylane-lightning/blob/522f3abd4a3e94eb542e961aae56cb81a93b40e1/pennylane_lightning/lightning_kokkos/lightning_kokkos.py#L472) and [LGPU](https://github.com/PennyLaneAI/pennylane-lightning/blob/522f3abd4a3e94eb542e961aae56cb81a93b40e1/pennylane_lightning/lightning_gpu/lightning_gpu.py#L561)
```suggestion
# Whether the compilation pass should optimize the first
# state preparation as a SetState operation.
skip_initial_state_prep = true
```

File: pennylane_lightning/lightning_gpu/lightning_gpu.toml [hunk 2]
```diff
@@ -109,3 +109,6 @@ dynamic_qubit_management = false
 # in a single execution
 non_commuting_observables = true
 
+# Whether the compilation pass should optimize the first
+# state preparation as a SetState operation.
+skip_initial_state_prep = false
```
Comment (maliasadi) [inline pennylane_lightning/lightning_gpu/lightning_gpu.toml:114]:
```suggestion
skip_initial_state_prep = true
```

File: pennylane_lightning/lightning_kokkos/lightning_kokkos.toml (hunk 1 above)
Comment (erick-xanadu) [reply pennylane_lightning/lightning_kokkos/lightning_kokkos.toml:117]: I'm keeping this one false until the PR in Catalyst is fixed for this. Then I'll update. (This PR will not be merged until the PR in Catalyst is finished)

**Comment by AmintorDusko** [comment]
Hey @erick-xanadu, could you please update your branch? (merge master)

File: pennylane_lightning/lightning_gpu/lightning_gpu.toml [hunk 3]
```diff
@@ -109,3 +109,6 @@ dynamic_qubit_management = false
 # in a single execution
 non_commuting_observables = true
 
+# Whether the compilation pass should optimize the first
+# state preparation as a SetState operation.
```
Comment (dime10) [inline pennylane_lightning/lightning_gpu/lightning_gpu.toml:113]: Let's not describe it as an optimization:
```suggestion
# Whether the device supports (arbitrary) initial state preparation.
```

File: pennylane_lightning/lightning_kokkos/lightning_kokkos.toml (hunk 1 above)
Comment (dime10) [reply pennylane_lightning/lightning_kokkos/lightning_kokkos.toml]: Not sure I understand the reasoning, but as long this is changed before merging.

**Review by dime10 — APPROVED** [review]
Thanks Erick!

File: pennylane_lightning/lightning_kokkos/lightning_kokkos.toml (hunk 1 above)
Comment (erick-xanadu) [reply pennylane_lightning/lightning_kokkos/lightning_kokkos.toml]: Done.

File: pennylane_lightning/lightning_gpu/lightning_gpu.toml (hunk 3 above)
Comment (erick-xanadu) [reply pennylane_lightning/lightning_gpu/lightning_gpu.toml]: Changed.

File: pennylane_lightning/lightning_kokkos/lightning_kokkos.toml (hunk 1 above)
Comment (erick-xanadu) [reply pennylane_lightning/lightning_kokkos/lightning_kokkos.toml]: https://github.com/PennyLaneAI/pennylane-lightning/pull/826/commits/efc57527628f89af56eeca7ea71b3278e284d510

File: pennylane_lightning/lightning_gpu/lightning_gpu.toml (hunk 3 above)
Comment (erick-xanadu) [reply pennylane_lightning/lightning_gpu/lightning_gpu.toml]: https://github.com/PennyLaneAI/pennylane-lightning/pull/826/commits/afe6d841a1977de4974d80ea143d574e24e70536

**Review by vincentmr — CHANGES_REQUESTED** [review]
LGTM, but please update the change log.

**Comment by erick-xanadu** [comment]
@vincentmr , done!

**Review by vincentmr — APPROVED** [review]
LGTM, cheers @erick-xanadu .

**Pull request merged**